File: Ticketing_Portal_Project/Backend/Libraries/TicketingLibrary/Models/Ticket.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace TicketingLibrary.Models
{
    [Table("Ticket")]
    public class Ticket
    {
        [Key]
        [Column(TypeName = "CHAR(4)")]
        public string? TicketId { get; set; }

        [Column(TypeName = "CHAR(15)")]
        public string? Title { get; set; }

        [Column(TypeName = "CHAR(100)")]
        public string? Description { get; set; }

        [Column(TypeName = "CHAR(4)")]
        public string? TicketTypeId { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime? TicketCreatedDate { get; set; }

        public string? Status { get; set; }

        [Column(TypeName = "CHAR(4)")]
        
        public string? CreatedByEmpId { get; set; }

        [Column(TypeName = "CHAR(4)")]
        
        public string? AssignedToEmpId { get; set; }

        // [JsonIgnore]
        [ForeignKey("TicketTypeId")]
        public virtual TicketType? TicketType { get; set; }

        // [JsonIgnore]
        [ForeignKey("CreatedByEmpId")]
        public virtual Employee? CreatedByEmployee { get; set; }

        // [JsonIgnore]
        [ForeignKey("AssignedToEmpId")]
        public virtual Employee? AssignedToEmployee { get; set; }

        // [JsonIgnore]
        public virtual ICollection<TicketReply> TicketReplies { get; set; } = new List<TicketReply>();
    }
}

File: Ticketing_Portal_Project/Frontend/angular-app/src/app/models/Ticket.ts
import { TicketType } from './TicketType';
import { Employee } from './Employee';

export class Ticket {
  public ticketId: string;
  public title: string;
  public description: string;
  public ticketTypeId: string;
  public ticketCreatedDate: Date;
  public status: string;
  public createdByEmpId: string;
  public assignedToEmpId: string;

  constructor(ticketId: string, title: string, description: string,ticketTypeId: string,ticketCreatedDate: Date,status: string,createdByEmpId: string,assignedToEmpId: string
  ) {
    this.ticketId = ticketId;
    this.title = title;
    this.description = description;
    this.ticketTypeId = ticketTypeId;
    this.ticketCreatedDate = ticketCreatedDate;
    this.status = status;
    this.createdByEmpId = createdByEmpId;
    this.assignedToEmpId = assignedToEmpId;
  }
}
